src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+                        // So far, ODL only suppports the string default value.
+                        defaultValue = propertyValue.ParseAsJson();
+                        break;
+
+                    default:
+                        // it MAY contain annotations.
+                        ParseCsdlAnnotation(propertyName, propertyValue, context, propertyAnnotations);
+                        break;
+                }
+            });
+
+            CsdlProperty csdlProperty = new CsdlProperty(name, typeReference, defaultValue, context.Location());
+            propertyAnnotations.ForEach(a => csdlProperty.AddAnnotation(a));
+            return csdlProperty;
+        }
+
+        /// <summary>
+        /// Parse the <see cref="JsonElement"/> to <see cref="CsdlNavigationProperty"/>.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <param name="navObject">The json value to parse from.</param>
+        /// <param name="jsonPath">The json path of this value.</param>
+        /// <returns>the parsed <see cref="CsdlNavigationProperty"/>.</returns>
+        internal static CsdlNavigationProperty ParseCsdlNavigationProperty(string name, JsonElement navObject, JsonParserContext context)
+        {
+            Debug.Assert(navObject.ValueKind == JsonValueKind.Object);
+            Debug.Assert(context != null);
+
+            // Navigation properties are represented as members of the object representing a structured type.
+            // The member name is the property name, the member value is an object.
+            // The navigation property object MUST contain the member $Kind with a string value of NavigationProperty.
+            ValidateKind(navObject, context, "NavigationProperty", required: true);
+
+            // It MUST contain the member $Type, and it MAY contain the members $Collection, $Nullable, $Partner, $ContainsTarget, $ReferentialConstraint, and $OnDelete.
+            string typeName = null;
+            bool? collection = null;
+            bool? nullable = null;
+            string parter = null;
+            bool? containsTarget = null; // Absence of the member means false.
+
+            IList<CsdlAnnotation> onDeleteAnnotations = new List<CsdlAnnotation>();
+            CsdlOnDelete csdlOnDelete = null;
+
+            IList<CsdlAnnotation> navPropertyAnnotations = new List<CsdlAnnotation>();
+            IList<CsdlReferentialConstraint> referentialConstraints = null;
+
+            navObject.ParseAsObject(context, (propertyName, propertyValue) =>
+            {
+                switch (propertyName)
+                {
+                    case "$Kind":
+                        break;
+
+                    case "$Type":
+                        // Absence of the $Type member means the type is Edm.String. This member SHOULD be omitted for string properties to reduce document size.
+                        typeName = propertyValue.ParseAsString(context);

```

Comment:
nav prop can't be a string, right?

Reply:
Navigation property type should be a string. The comment at line 768 is for general information of $Type. 
Let me update the comments.

---
In reply to: [443078502](https://github.com/OData/odata.net/pull/1775#discussion_r443078502) [](ancestors = 443078502)